addon/services/engine-router-service.js

```diff
@@ -0,0 +1,121 @@
+import Ember from 'ember';
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { computed } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+
+// @HACK: since `getEngineParent` is not exported
+function getEngineParent(engine) {
+  const symbolPrefix = `__ENGINE_PARENT${Ember.GUID_KEY}`;
+  const symbol = Object.keys(engine).find(k => k.startsWith(symbolPrefix));
+  if (!symbol) {
+    return null;
+  }
+  return engine[symbol];
+}
+
+export default Service.extend({
+  engine: computed(function () {
+    return getOwner(this);
+  }),
+
+  externalRoutes: reads('engine._externalRoutes'),
+
+  mountPoint: reads('engine.mountPoint'),
+
+  rootURL: reads('externalRouter.rootURL'),
+
+  currentURL: reads('externalRouter.currentURL'),
+
+  currentRouteName: computed('externalRouter.currentRouteName', function () {
+    if (this.externalRouter.currentRouteName === this.mountPoint) {
+      return 'application';
+    }
+    return this.externalRouter.currentRouteName.slice(this.mountPoint.length + 1);
+  }),
+
+  externalRouter: computed(function () {
```

Comment:
We should use our existing service aliasing architecture to make the host applications `router` service show up in a different location (e.g. `service:-root-router'`) and then change this to:

```js
_rootRouteService: inject('-root-router'),
```

This will allow us to remove the `getEngineParent` code in this PR AFAICT

Reply:
@rwjblue I don't get your point sorry, how can I show up this `router` service in a different location?

Reply:
solved